src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -514,9 +514,7 @@ public void SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging()
                         {
                             // Forces a synchronous exception for WinHttpHandler
```

Comment:
The comment is correct, however, the test code I modified is not correct.

As Karel mentioned above, the test is to force an exception during SendAsync/GetAsync: we want to observe/log the exception, and wait for the current task get faulted/stopped.

Before the change in WinHttpHandler, setting `CookieContainer` to null will result in `InvalidOperationException` in `GetAsync`. But after the change, the exception will be immediately thrown at the setter.

My current modification in test result in the test failure, because there is no `SendAsync/GetAsync` activity for PlatformHandler . I will try to force a difference Exception (from SendAsync/GetAsync) for PlatformHandler in this test.

Reply:
>Before the change in WinHttpHandler, setting CookieContainer to null 

I thought you were only changing HttpClientHandler for this fix??

Reply:
Sorry wrong wording. 

What I mean is before the change in HttpClientHandle (a layer above WinHttpHandler), setting CookieContainer to null will result in InvalidOperationException in GetAsync for WinHttpHandler.

Since I changed HttpClientHandle to check if CookieContainer is null, and throw Exception. This prevents the Exception from WinHttpHandler SendAsync.